### PR TITLE
feat(schemas): add migrations table

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add a new table `migrations`, prepare to be used for migrations.

The table name is started with underscore `_` since it is a special table and irrelevant to other business.

Brief introduction of the plan on migrations:

1. We'll create a migration file each time we change our db structure. This file will contain JS code to change the db structure and migrate data if needed.
2. Compare files and db's `_migrations` table, find the migration files need to run for the current db.
3. Run the files.

The migration steps MUST run with exact order, so add checksums to ensure that the history file is not been modifid.

### CODE REVIEW

File: packages/schemas/tables/migrations.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+create table _migrations (
```
Comment (demonzoo) [inline packages/schemas/tables/migrations.sql:1]: nit: Just asking, maybe single form is OK? Say `_migration`

**Comment by demonzoo** [comment]
What data and when will this data be persisted in this table?

File: packages/schemas/tables/migrations.sql (hunk 1 above)
Comment (wangsijie) [reply packages/schemas/tables/migrations.sql:1]: But all other table names are in plural.

**Comment by wangsijie** [comment]
Assume that we have 3 new migration files:
1. 1658921424_add_user_created_at
2. 1658922424_add_application_secret
3. 1658923424_change_app_id_to_client_id

Then when the migration begins, these 3 files will be run in sequence, and inserting 3 rows in the table.

And next time, we can skip those files according the this table's rows.

@demonzoo FYI

File: packages/schemas/tables/migrations.sql (hunk 1 above)
Comment (demonzoo) [reply packages/schemas/tables/migrations.sql:1]: ok, np

File: packages/schemas/tables/migrations.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  id varchar(128) not null,
+  name varchar(256) not null unique,
+  checksum varchar(64) not null,
+  started_at timestamptz not null default(now()),
+  finished_at timestamptz,
```
Comment (demonzoo) [inline packages/schemas/tables/migrations.sql:6]: nit: completedAt

**Review by demonzoo — APPROVED** [review]

File: packages/schemas/tables/migrations.sql (hunk 1 above)
Comment (gao-sun) [inline packages/schemas/tables/migrations.sql:1]: let's discuss offline - i didn't get the point of this table

**Pull request closed without merging**